server/src/lib/jobs/resolve.cpp

```diff
@@ -396,6 +396,54 @@ static std::string byHostedBy(const Group::Condition& cond)
 
 static std::string byGroupId(fty::db::Connection& conn, const Group::Condition& cond);
 
+static std::string byTag(const Group::Condition& cond)
+{
+    static std::string sql = R"(
+        SELECT DISTINCT r.id_asset_element 
+        FROM t_bios_asset_element_tag_relation AS r 
+        INNER JOIN t_bios_tag AS tag 
+        ON r.id_tag = tag.id_tag 
+        WHERE tag.name {op} '{val}'
+    )";
+
+    // clang-format off
+    std::string ret = fmt::format(sql,
+        "op"_a  = op(cond),
+        "val"_a = value(cond)
+    );
+    // clang-format on
+
+    static std::string sqlAdditional = R"(
+        {sql}
+        AND r.id_asset_element NOT IN (
+            SELECT r.id_asset_element 
+            FROM t_bios_asset_element_tag_relation AS r 
+            INNER JOIN t_bios_tag AS tag 
+            ON r.id_tag = tag.id_tag 
+            WHERE tag.name {op} '{val}'
+        )
+    )";
+
+    std::string opAdditional;
+    if (cond.op == Group::ConditionOp::IsNot) {
+        opAdditional = "=";
+    } else if (cond.op == Group::ConditionOp::DoesNotContain) {
```

Comment:
'Does not contain' and 'like', this give sense in whole sql context, but looks very weird :)
And condition below... 
mmm...
You could make just one if scope for negotiation.... otherwise it looks like spaghetti. :)